### PR TITLE
upgrade scalacheck and use CoGen to generate random functions

### PR DESCRIPTION
**Pull request by julien-truffaut**
see https://github.com/rickynils/scalacheck/issues/241

### CODE REVIEW

**Comment by julien-truffaut** [comment]
waiting for https://github.com/rickynils/scalacheck/pull/242

**Comment by julien-truffaut** [comment]
It should be good for review

File: build.sbt [hunk 1]
```diff
@@ -47,10 +47,10 @@ lazy val shapeless  = Def.setting("com.chuusai"     %%% "shapeless"   % "2.3.0")
 
 lazy val refinedVersion = "0.4.0"
 lazy val refinedDep = Def.setting("eu.timepit"      %%% "refined"     % refinedVersion)
-lazy val refinedScalaCheckDep = Def.setting("eu.timepit"  %%% "refined-scalacheck" % refinedVersion % "test")
 
-lazy val discpline  = Def.setting("org.typelevel"   %%% "discipline"  % "0.4")
-lazy val scalatest  = Def.setting("org.scalatest"   %%% "scalatest"   % "3.0.0-M7"  % "test")
+lazy val discpline  = Def.setting("org.typelevel"   %%% "discipline"  % "0.5")
```
Comment (aoiroaoino) [inline build.sbt]: typo val name?

Comment (julien-truffaut) [inline build.sbt]: good catch, I will change that

**Comment by aoiroaoino** [comment]
LGTM 👍

**Pull request merged**